omnigibson/robots/manipulation_robot.py

```diff
@@ -1073,24 +1087,20 @@ def _handle_assisted_grasping(self, action):
             assert cmd_dim == 1, \
                 f"Gripper {arm} controller command dim must be 1 to use assisted grasping, got: {cmd_dim}."
 
-            # TODO: Why are we separately checking for complementary conditions?
-            threshold = np.mean(self._controllers[f"gripper_{arm}"].command_input_limits)
-            applying_grasp = action[self.controller_action_idx[f"gripper_{arm}"][0]] < threshold
-            releasing_grasp = action[self.controller_action_idx[f"gripper_{arm}"][0]] > threshold
+            controller = self._controllers[f"gripper_{arm}"]
+            controlled_joints = controller.dof_idx
+            threshold = np.mean(np.array(self.control_limits["position"])[:, controlled_joints], axis=0)
+            applying_grasp = np.any(controller.control < threshold)
```

Comment:
We are intentionally switching to the control here so that when holding with 0 delta, the side of the threshold doesnt change. The any is such that as long as 1 finger is closed enough, this will start/remain grasping.